extensions/panache/hibernate-orm-panache-rest/deployment/src/main/java/io/quarkus/hibernate/orm/panache/rest/deployment/HibernateOrmPanacheRestProcessor.java

```diff
@@ -0,0 +1,96 @@
+
+import static io.quarkus.deployment.builditem.FeatureBuildItem.HIBERNATE_ORM_PANACHE_REST;
+
+import java.lang.reflect.Modifier;
+import java.util.Arrays;
+
+import org.jboss.jandex.ClassInfo;
+import org.jboss.jandex.DotName;
+import org.jboss.jandex.IndexView;
+import org.jboss.logging.Logger;
+import org.jboss.resteasy.links.impl.EL;
+
+import io.quarkus.arc.deployment.GeneratedBeanBuildItem;
+import io.quarkus.arc.deployment.GeneratedBeanGizmoAdaptor;
+import io.quarkus.deployment.annotations.BuildProducer;
+import io.quarkus.deployment.annotations.BuildStep;
+import io.quarkus.deployment.builditem.CombinedIndexBuildItem;
+import io.quarkus.deployment.builditem.FeatureBuildItem;
+import io.quarkus.deployment.builditem.nativeimage.RuntimeInitializedClassBuildItem;
+import io.quarkus.hibernate.orm.panache.rest.PanacheCrudController;
+import io.quarkus.hibernate.orm.panache.rest.runtime.hal.HalCollectionWrapper;
+import io.quarkus.hibernate.orm.panache.rest.runtime.hal.HalCollectionWrapperJacksonSerializer;
+import io.quarkus.hibernate.orm.panache.rest.runtime.hal.HalCollectionWrapperJsonbSerializer;
+import io.quarkus.hibernate.orm.panache.rest.runtime.hal.HalEntityWrapper;
+import io.quarkus.hibernate.orm.panache.rest.runtime.hal.HalEntityWrapperJacksonSerializer;
+import io.quarkus.hibernate.orm.panache.rest.runtime.hal.HalEntityWrapperJsonbSerializer;
+import io.quarkus.hibernate.orm.panache.rest.runtime.hal.HalLink;
+import io.quarkus.hibernate.orm.panache.rest.runtime.hal.HalLinkJacksonSerializer;
+import io.quarkus.hibernate.orm.panache.rest.runtime.hal.HalLinkJsonbSerializer;
+import io.quarkus.jackson.spi.JacksonModuleBuildItem;
+import io.quarkus.jsonb.spi.JsonbSerializerBuildItem;
+
+class HibernateOrmPanacheRestProcessor {
+
+    private static final DotName PANACHE_CRUD_CONTROLLER_INTERFACE = DotName
+            .createSimple(PanacheCrudController.class.getName());
+
+    private static final Logger LOGGER = Logger.getLogger(HibernateOrmPanacheRestProcessor.class);
+
+    @BuildStep
+    FeatureBuildItem feature() {
+        return new FeatureBuildItem(HIBERNATE_ORM_PANACHE_REST);
+    }
+
+    @BuildStep
+    RuntimeInitializedClassBuildItem el() {
+        return new RuntimeInitializedClassBuildItem(EL.class.getCanonicalName());
+    }
+
+    @BuildStep
+    void generateControllers(CombinedIndexBuildItem index, BuildProducer<GeneratedBeanBuildItem> implementationsProducer) {
+        ControllerImplementor implementor = new ControllerImplementor(new GeneratedBeanGizmoAdaptor(implementationsProducer));
+        for (ClassInfo classInfo : index.getIndex().getKnownDirectImplementors(PANACHE_CRUD_CONTROLLER_INTERFACE)) {
+            if (isValidCrudController(index.getIndex(), classInfo)) {
+                implementor.implement(classInfo, index.getIndex());
+            }
+        }
+    }
+
+    private boolean isValidCrudController(IndexView index, ClassInfo classInfo) {
```

Comment:
Why not throwing and exception instead of logging in WARN ?
After all, it will not work as expected so better to throw ?

Reply:
Could do that if it's a more appropriate for Quarkus.

Reply:
Yeah, I think failing the build would be more appropriate. We do that in other places, like Spring Data JPA.